src/__tests__/json-schema.test.js

```diff
@@ -0,0 +1,266 @@
+describe('json-schema', () => {
+  const Sequelize = require('sequelize')
+  const sequelize = new Sequelize('postgres://fakeUser:pass@fake.com:1111/fake')
+  const SEQUELIZE_JSON_TABLE = [
+    [
+      'uuid',
+      Sequelize.UUID,
+      {
+        type: 'string',
+        format: 'uuid'
+      }
+    ],
+    [
+      'string',
+      Sequelize.STRING,
+      { type: 'string' }
+    ],
+    [
+      'string 123',
+      Sequelize.STRING(1234),
+      { type: 'string' }
+    ],
+    [
+      'text',
+      Sequelize.TEXT,
+      { type: 'string' }
+    ],
+    [
+      'integer',
+      Sequelize.INTEGER,
+      { type: 'integer' }
+    ],
+    [
+      'integer',
+      Sequelize.BIGINT,
+      { type: 'integer' }
+    ],
+    [
+      'integer 11',
+      Sequelize.BIGINT(11),
+      { type: 'integer' }
+    ],
+    [
+      'float',
+      Sequelize.FLOAT,
+      { type: 'number' }
+    ],
+    [
+      'float 11',
+      Sequelize.FLOAT(11),
+      { type: 'number' }
+    ],
+    [
+      'real',
+      Sequelize.REAL,
+      { type: 'number' }
+    ],
+    [
+      'real 11',
+      Sequelize.REAL(11),
+      { type: 'number' }
+    ],
+    [
+      'double',
+      Sequelize.DOUBLE,
+      { type: 'number' }
+    ],
+    [
+      'double 11',
+      Sequelize.DOUBLE(11),
+      { type: 'number' }
+    ],
+    [
+      'decimal',
+      Sequelize.DECIMAL,
+      { type: 'number' }
+    ],
+    [
+      'decimal 10,2',
+      Sequelize.DECIMAL(10, 2),
+      { type: 'number' }
+    ],
+    [
+      'date',
+      Sequelize.DATE,
+      { type: 'date' }
+    ],
+    [
+      'date',
+      Sequelize.DATEONLY,
+      { type: 'date' }
+    ],
+    [
+      'boolean',
+      Sequelize.BOOLEAN,
+      { type: 'boolean' }
+    ],
+    [
+      'blob',
+      Sequelize.BLOB,
+      { type: 'string' }
+    ],
+    [
+      'array text',
+      Sequelize.ARRAY(Sequelize.TEXT),
+      { type: 'array' }
+    ]
+  ]
+
+  beforeEach(() => {
+    jest.mock('uuid/v4', () => ({
+      __esModule: true,
+      default: () => '0ef9f424-c6f0-4bfe-b55e-87c50420180f'
+    }))
+  })
+
+  afterEach(() => {
+    jest.resetModules()
+    jest.clearAllMocks()
+  })
+
+  test.each(SEQUELIZE_JSON_TABLE)(
+    'should handle type %s',
+    (type, _sequelize, json) => {
+      const { jsonSchema } = require('../json-schema')
+      const model = sequelize.define(type, {
+        id: {
+          type: _sequelize,
+          primaryKey: true
```

Comment:
How come there is a primary key on each of these?

Reply:
removed